Update release notes refactoring (#1368)
demisto/demisto-sdk: CHANGELOG.md

* Added new validation that DBotRole is set for scripts that requires elevated permissions to the `XSOAR-linter` in the **lint** command.
* Added url escaping to markdown human readable section in generate docs to avoid autolinking.
* Added a validation that mapper's id and name are matching. Updated the format of mapper to include update_id too.
* Refactored the **update-release-notes** command:

>   - Replaced the *--all* flag with *-g*

Author: Done

* Added a validation that mapper's id and name are matching. Updated the format of mapper to include update_id too.
* Added a validation to ensure that image paths in the README files are valid.
* Fixed **find_type** function to correctly find test files, such as, test script and test playbook.
* Refactored the **update-release-notes** command:
  - Replaced the *--all* flag with *--use-git* or *-g*.
  - Added the *--force* flag to update the pack release notes without changes in the pack.
  - The **update-release-notes** command will now update all dependent integrations on ApiModule change, even if not specified.
  - If more than one pack has changed, the full list of updated packs will be printed at the end of **update-release-notes** command execution.

>   - Fixed an issue where **update-release-notes** command did not add docker image release notes entry for release notes file if a script was changed.

Reviewer: ```suggestion
  - Fixed an issue where the **update-release-notes** command did not add a docker image entry for the release notes file if a script was changed.
```

>   - Fixed an issue where **update-release-notes** command did not detect changed files which had the same name.
>   - Fixed an issue where **update-release-notes** command did not support fromVersion field in JSON files.

Reviewer: ```suggestion
  - Fixed an issue where the **update-release-notes** command did not detect changed files that had the same name.
  - Fixed an issue in the **update-release-notes** command where the version support of JSON files was mishandled.
```

* Added a version differences section to readme in **generate-docs** command.
* Added the *--docs-format* flag in the **integration-diff** command to get the output in README format.
* Added the *--input-old-version* and *--skip-breaking-changes* flags in the **generate-docs** command to get the details for the breaking section and to skip the breaking changes section.
* Refactor **update-release-notes** command - new classs manager, split some logic to smaller functions, added docsrings and type annotations.

> * Added a  *--force* flag to the **update-release-notes** command, so the update can take place even without a change in the pack.

Author: Done, please see additions.
